napari/layers/base/base.py

```diff
@@ -291,6 +301,48 @@ def __repr__(self):
         cls = type(self)
         return f"<{cls.__name__} layer {repr(self.name)} at {hex(id(self))}>"
 
+    def _mode_setter_helper(self, mode, Modeclass):
```

Comment:
Can you add `Parameters` to doc string here too. `Modeclass` i guess is the layer specific enum of modes available to that layer.

Now that more mode stuff is getting added to the base, I wonder if we shouldn't define the `Mode` enum just "once" with all possible modes and then have another way that layers can decide which modes they want to access, instead of right now having both https://github.com/napari/napari/blob/a9b0e68055d6fb880fa8a019d2842dd3f6d42ad6/napari/layers/points/_points_constants.py#L24-L36 and https://github.com/napari/napari/blob/a9b0e68055d6fb880fa8a019d2842dd3f6d42ad6/napari/layers/labels/_labels_constants.py#L9-L37 etc etc

That could come in a future PR, but just flagging

Reply:
Done for docstrings. 

And yes for Mode, though not all Layers have Modes (AFAICT), and the modes have different values. We can make Mode a class attribute, but I was afraid that would make it confusing that all layers have a mode. 

I guess we could make all layer have modes, even if they have a single variant...

Reply:
> And yes for Mode, though not all Layers have Modes (AFAICT), and the modes have different values. We can make Mode a class attribute, but I was afraid that would make it confusing that all layers have a mode.

Yeah, that's what I'm worried about too. I guess I'd make "Pan/ Zoom" mode the default mode so that all layers have that mode and by default start out there. Other layers can then add additional modes as desired. But this can come in a later PR if we think it simplifies things